api/backend/blueflood/blueflood.go

```diff
@@ -72,11 +72,50 @@ func (b *Blueflood) Api() api.API {
 }
 
 func (b *Blueflood) FetchSeries(metric api.TaggedMetric, predicate api.Predicate, sampleMethod api.SampleMethod, timerange api.Timerange) (*api.SeriesList, error) {
-	graphiteMetric, err := b.api.ToGraphiteName(metric)
+	metricTagSets, err := b.Api().GetAllTags(metric.MetricKey)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: Be a little smarter about this?
+	resultSeries := []api.Timeseries{}
+	for _, ts := range metricTagSets {
+		if predicate == nil || predicate.Apply(ts) {
```

Comment:
i think we should pass around all-matching predicate rather than `nil` predicate. Empty object pattern :P